tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -113,21 +113,37 @@ internal static string PropagateDataViaComment(DbmPropagationLevel propagationSt
         /// Currently only working for MSSQL (uses an instruction that is specific to it)
         /// </summary>
         /// <returns>True if the traceparent information was set</returns>
-        internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbConnection? connection, Span span)
+        internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbCommand command, Span span)
         {
-            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || connection == null)
+            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient)
             {
                 return false;
             }
 
+            // NOTE: For Npgsql command.Connection throws NotSupportedException for NpgsqlDataSourceCommand (v7.0+)
+            //       Since the feature isn't available for Npgsql we avoid this due to the integrationId check above
+            if (command.Connection == null)
+            {
+                return false;
+            }
+
+            if (command.Connection.State != ConnectionState.Open)
+            {
+                Log.Debug("PropagateDataViaContext did not have an Open connection, so it could not propagate Span data for DBM. Connection state was {ConnectionState}", command.Connection.State);
+
+                return false;
+            }
+
             var stopwatch = System.Diagnostics.Stopwatch.StartNew();
 
             const byte version = 0; // version can have a maximum value of 15 in the current format
             var sampled = SamplingPriorityValues.IsKeep(span.Context.TraceContext.GetOrMakeSamplingDecision());
```

Comment:
Note part of this PR, but are we ok forcing a sampling decision here? What is the sampling decision in the injected context used for? Does it affect sampling of other data aside from the trace (i.e. DBM data)? Is it propagated to other services?

We may want to use the sampling decision _if there is one already_, but otherwise not force it to happen yet, since we generally want to delay it until it's required for downstream propagation or when sending the trace to the agent.

Reply:
Hmm, I think the DBM context doesn't support "no sampling decision yet," so never mind, I guess.

Reply:
yes it may be worth asking them, I don't think they do anything with the sampling decision so far, it's just written to do it like papa (the other propagation mechanisms)

Reply:
papa?

Reply:
I mean take example on what's here before. It's a metaphor ^^